tools/test.py

```diff
@@ -255,6 +256,9 @@ def HasRun(self, output):
         logger.info('#' + l)
       for l in output.output.stdout.splitlines():
         logger.info('#' + l)
+    elif output.HasSkipped():
+      skip = is_skipped.findall(output.output.stdout)      
```

Comment:
Is there a reason to use `.findall()` instead of `.search()` here?  I would just drop HasSkipped() and write it like this:

```python
else:
  match = re.match(r'# SKIP\S*\s+(.*)', output.output.stdout, re.IGNORECASE)
  if match:
    logger.info('ok %i - %s # skip %s' % (self._done, command, match.group(1)))
  else:
    logger.info('ok %i - %s' % (self._done, command))
```

Minor aside: if you want to keep the regex separate, maybe rename `is_skipped` to e.g. `skip_regex`.